spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestFilteredWithSystemFunctionScan.java

```diff
@@ -0,0 +1,784 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.source;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+import java.time.LocalDate;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.spark.Spark3Util;
+import org.apache.iceberg.spark.SparkTestBaseWithCatalog;
+import org.apache.iceberg.spark.functions.BucketFunction;
+import org.apache.iceberg.spark.functions.DaysFunction;
+import org.apache.iceberg.spark.functions.MonthsFunction;
+import org.apache.iceberg.spark.functions.TruncateFunction;
+import org.apache.iceberg.spark.functions.YearsFunction;
+import org.apache.spark.sql.connector.expressions.Expression;
+import org.apache.spark.sql.connector.expressions.FieldReference;
+import org.apache.spark.sql.connector.expressions.LiteralValue;
+import org.apache.spark.sql.connector.expressions.UserDefinedScalarFunc;
+import org.apache.spark.sql.connector.expressions.filter.And;
+import org.apache.spark.sql.connector.expressions.filter.Not;
+import org.apache.spark.sql.connector.expressions.filter.Or;
+import org.apache.spark.sql.connector.expressions.filter.Predicate;
+import org.apache.spark.sql.connector.read.Batch;
+import org.apache.spark.sql.connector.read.ScanBuilder;
+import org.apache.spark.sql.connector.read.SupportsPushDownV2Filters;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestFilteredWithSystemFunctionScan extends SparkTestBaseWithCatalog {
```

Comment:
Can this be added to a different test case? New test cases are expensive because they start catalogs and a Spark env. If there's another where this fits it would be better.

Reply:
What about adding it to `TestFilteredScan`? But this could make `TestFilteredScan` very large.

Reply:
That wouldn't really help because `TestFilteredScan` isn't a `SparkTestBaseWithCatalog` test. Would `TestSelect` work?

Reply:
@rdblue `SparkScanBuilder` is only accessible under package `org.apache.iceberg.spark.source`. So maybe `TestSparkScan` is a good choice.

Reply:
`TestSparkScan` sounds reasonable to me.